### PR TITLE
[v3.8.0] feat: new property decorator `radioGroup` & improve properties rendering

### PR DESCRIPTION
**Pull request by doommm**
Re: 
- https://github.com/cocos/3d-tasks/issues/8715
- https://github.com/cocos/3d-tasks/issues/12255

### Changelog

- For enum type data, it provides the ability to control the ui component(select, button) used in inspector
image
- improve label component rendering in editor inspector. use `ui-radio-group` and `ui-radio-button` instead of `ui-tab`
image

### Depends
- https://github.com/cocos/creator-ui-kit/pull/320

#### Button

This will render entire enumList to a Radio Group(With Radio Button inside).

```ts
@type(ModelShadowCastingMode)
@radioGroup
get shadowCastingMode () {
    return this._shadowCastingMode;
}
```

image

image

#### Select

Default behavior. Same As before. The `radioGroup` decorator is not necessary.

```ts
// it will render as a 'select' component implicitly
@type(ModelShadowCastingMode)
get shadowCastingMode () {
    return this._shadowCastingMode;
}
```

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/3d/framework/mesh-renderer.ts [hunk 1]
```diff
@@ -370,6 +370,7 @@ export class MeshRenderer extends ModelRenderer {
     @tooltip('i18n:model.shadow_casting_model')
```
Comment (jareguo) [inline cocos/3d/framework/mesh-renderer.ts:370]: 和下面重复了

File: cocos/3d/framework/mesh-renderer.ts [hunk 2]
```diff
@@ -379,6 +380,16 @@ export class MeshRenderer extends ModelRenderer {
         this._updateCastShadow();
     }
 
+    @tooltip('i18n:model.shadow_casting_model')
+    @group({ id: 'DynamicShadow', name: 'DynamicShadowSettings' })
+    @disallowAnimation
+    get shadowCastingModeBool (): boolean {
```
Comment (jareguo) [inline cocos/3d/framework/mesh-renderer.ts:386]: prefer _shadowCastingModeForInspector

File: cocos/core/data/class.ts [hunk 3]
```diff
@@ -517,6 +517,7 @@ function parseAttributes (constructor: Function, attributes: PropertyStash, clas
         parseSimpleAttribute('slide', 'boolean');
         parseSimpleAttribute('unit', 'string');
         parseSimpleAttribute('userData', 'object');
+        parseSimpleAttribute('radioGroup', 'boolean');
```
Comment (jareguo) [inline cocos/core/data/class.ts:520]: 在哪有用到？

Comment (doommm) [reply cocos/core/data/class.ts:520]: 目前还没有地方使用。需要在这个 PR 里进行修改，还是在合并后另外开 PR ？

File: cocos/3d/framework/mesh-renderer.ts (hunk 1 above)
Comment (doommm) [reply cocos/3d/framework/mesh-renderer.ts:370]: 不在 inspector 中显示的话，tooltip, group, disallowAnimation 这些装饰器是不是都可以去掉

File: cocos/core/data/utils/attribute-defines.ts [hunk 4]
```diff
@@ -147,6 +147,11 @@ export interface IExposedAttributes {
      * @zh 用户自定义数据，可以通过 `CCClass.attr()` 接口获取自定义数据。
      */
     userData?: Record<string, any>;
+
+    /**
+     * 在允许的情况下，在编辑器中显示为一组单选按钮
```
Comment (arsen2010) [inline cocos/core/data/utils/attribute-defines.ts:152]: 如果有注释需要添加 `@en` 和 `@zh` ,这样会被生成为对外文档

File: editor/inspector/components/label.js [hunk 5]
```diff
@@ -49,6 +49,42 @@ ui-tab {
...
+/**
+ * 
+ * @param {object} options
+ * @param {any[]} options.enumList
+ * @param {string} options.tooltip
+ * @param {(elementName: string) => string}options.getIconName
+ * @param {(event: CustomEvent) => {}} options.onChange
+ * @returns
+ */
+function createRadioGroup(options) {
```
Comment (arsen2010) [inline editor/inspector/components/label.js:61]: 这个可以通用方法可以提取到 '../utils/prop' 文件里后续复用

File: editor/inspector/components/label.js [hunk 6]
```diff
@@ -49,6 +49,42 @@ ui-tab {
...
+        onChange(e);
+    });
+
+    for (let index = 0; index < enumList.length; index++) {
+        const element = enumList[index];
+        const icon = document.createElement('ui-icon');
+        const button = document.createElement('ui-radio-button');
+
+        const iconName = getIconName(element.name);
+        const tooltip = `${rawTooltip}_${element.name.toLocaleLowerCase()}`;
```
Comment (arsen2010) [inline editor/inspector/components/label.js:75]: tooltip 可能不够明确，存在两种情况的 tooltip：
1、ui-radio-group 上的提示，移到任一 ui-radio-button 都显示同一个；
2、每个 ui-radio-button 独立的 tooltip ，此时耦合 enumName 是可以的。
所以是不是加一个 option.enumTooltip = true 入参来区分两种情况

File: editor/inspector/components/label.js [hunk 7]
```diff
@@ -49,6 +49,42 @@ ui-tab {
...
+
+        icon.value = iconName;
```
Comment (arsen2010) [inline editor/inspector/components/label.js:77]: 如果 createRadioGroup 作为通用方法的话，button 的 innerHTML 需要来自 option.createButtonHTML() 方法，类似 getIconName

File: editor/inspector/components/label.js (hunk 5 above)
Comment (doommm) [reply editor/inspector/components/label.js:61]: 这个方法不是通用的。目前仅用在 label 这里

File: editor/inspector/components/label.js (hunk 6 above)
Comment (doommm) [reply editor/inspector/components/label.js:75]: 目前的垂直对齐、水平对齐只存在针对每个 radio-button 的 tooltip。

File: editor/inspector/components/label.js (hunk 7 above)
Comment (doommm) [reply editor/inspector/components/label.js:77]: createRadioGroup 暂不作为通用方法

File: cocos/core/data/utils/attribute-defines.ts (hunk 4 above)
Comment (doommm) [reply cocos/core/data/utils/attribute-defines.ts:152]: 上面的很多装饰器并未使用这个格式的注释

**Review by arsen2010 — APPROVED** [review]

**Comment by doommm** [comment]
@VisualSJ

**Review by VisualSJ — APPROVED** [review]

**Pull request merged**